sn_networking/src/driver.rs

```diff
@@ -695,6 +709,30 @@ impl SwarmDriver {
         let mut relay_manager_reservation_interval = interval(RELAY_MANAGER_RESERVATION_INTERVAL);
 
         loop {
+            // Prioritise any local cmds pending.
+            // https://github.com/libp2p/rust-libp2p/blob/master/docs/coding-guidelines.md#prioritize-local-work-over-new-work-from-a-remote
+            match self.local_cmd_receiver.try_recv() {
```

Comment:
the tokio_select will completes when `any of the branch completes`, which means after this `try_recv of local_cmd` , the next handling of `local_cmd` HAVE TO wait till any of the select branch completed.
which is not ideal for the case when there are bunch of local_cmds need to be handled ?

I'd image the next tokio_select will have to be in the style of `try_recv` as well, to make `local_cmds` really prioritized ?

Reply:
@maqi we have https://github.com/maidsafe/safe_network/pull/1981/files/01ca2b9111f4ddfb8d4ca165b324d8e7c9d13ade#diff-985b0f4f68bd09d2bcb4c2da67ab65c5c68d38d8aac388691e361a295bc8ae4bR723 

So when we get a `LocalSwarmCmd` successfully, we process it first and go to the next loop iter. Which, if there are local cmds, will be processed again. It's only if we `Err` there (ie, there are no more local cmds waiting) that we go to the `select!` macro.

Reply:
That said, I just reread the `select!` docs and it offers `biased` mode. Which actually I think makes this simpler/cleaner and offers us a way to process locally generated network activity `NetworkSwarmCmds` before incoming `SwarmEvent`s, which should also be beneficial.